witness/src/Dscp/Snowdrop/AccountValidation.hs

```diff
@@ -112,89 +119,58 @@ preValidateSimpleMoneyTransfer =
         -- Strip Update ctor from values.
         changes <- mapM checkThatAccountIsUpdatedOnly accountChanges
 
-        let (payer, recipients) = List.partition (is author) changes
+        let isAuthor :: (AccountId, Account) -> Bool
+            isAuthor (accId, _) = accId == author
+        let (payer, recipients) = List.partition isAuthor changes
+
+        allOutputs <- mapM validateSaneArrival recipients
 
-        (paid, deltas) <-
-            (,) <$> validateSaneDeparture payer before
-                <*> mapM validateSaneArrival recipients
+        let balanceBefore = aBalance before
+        -- Total amount of coins sent (sum of outputs)
+        let receivedTotal = sum allOutputs
+        -- Previous value minus fees
+        let receivedNoFees = receivedTotal - coinToInteger (unFees fees)
 
-        validateIff SumMustBeNonNegative $
-            paid - sum deltas == 0
+        unless (balanceBefore - receivedNoFees >= 0) $ throwLocalError SumMustBeNonNegative
+        unless (balanceBefore - receivedTotal >= 0) $ throwLocalError CannotAffordFees
 
+        validateSaneDeparture payer before
 
 -- | Require that whole projects into part or throw error.
 requirePart :: (HasReview e e1, MonadError e m, HasPrism s hash) => s -> e1 -> m hash
-requirePart whole message = do
-    case proj whole of
-      Just part -> do
-        return part
-
-      Nothing -> do
-        throwLocalError message
+requirePart whole message = maybe (throwLocalError message) pure $ proj whole
 
 -- | Require that id exists in a database or throw error.
-assertExists
-    :: ( HasExceptions e [e1, StatePException]
+assertExists ::
+       ( HasExceptions e '[ e1, StatePException]
```

Comment:
&lt;style-war&gt;
Why carrying first operator there, if all following operators are in the starts of their lines?
&lt;/style-war&gt;
Why manually lifting the list? I thought it worked fine w/o `'`.

Reply:
This was your change.